lambdas/data-migration2/tests/test-executions.js

```diff
@@ -0,0 +1,456 @@
+  process.env.ExecutionsTable = cryptoRandomString({ length: 10 });
+  process.env.AsyncOperationsTable = cryptoRandomString({ length: 10 });
+  process.env.CollectionsTable = cryptoRandomString({ length: 10 });
+  process.env.RulesTable = cryptoRandomString({ length: 10 });
+
+  await createBucket(process.env.system_bucket);
+
+  executionsModel = new Execution();
+  asyncOperationsModel = new AsyncOperation({
+    stackName: process.env.stackName,
+    systemBucket: process.env.system_bucket,
+  });
+  collectionsModel = new Collection();
+  rulesModel = new Rule();
+
+  await executionsModel.createTable();
+  await asyncOperationsModel.createTable();
+  await collectionsModel.createTable();
+  await rulesModel.createTable();
+
+  t.context.knexAdmin = await getKnexClient({
+    env: {
+      ...localStackConnectionEnv,
+      migrationDir,
+    },
+  });
+  await t.context.knexAdmin.raw(`create database "${testDbName}";`);
+  await t.context.knexAdmin.raw(`grant all privileges on database "${testDbName}" to "${testDbUser}"`);
+
+  t.context.knex = await getKnexClient({
+    env: {
+      ...localStackConnectionEnv,
+      PG_DATABASE: testDbName,
+      migrationDir,
+    },
+  });
+
+  await t.context.knex.migrate.latest();
+});
+
+test.afterEach.always(async (t) => {
+  await t.context.knex('executions').del();
+  await t.context.knex('collections').del();
+  await t.context.knex('async_operations').del();
+});
+
+test.after.always(async (t) => {
+  await executionsModel.deleteTable();
+  await asyncOperationsModel.deleteTable();
+  await collectionsModel.deleteTable();
+  await rulesModel.deleteTable();
+  await recursivelyDeleteS3Bucket(process.env.system_bucket);
+  await t.context.knex.destroy();
+  await t.context.knexAdmin.raw(`drop database if exists "${testDbName}"`);
+  await t.context.knexAdmin.destroy();
+});
+
+test.serial('migrateExecutionRecord correctly migrates execution record', async (t) => {
+  // This will be the top-level execution (no parent execution)
+  const fakeExecution = fakeExecutionFactoryV2({ parentArn: undefined });
+  const fakeCollection = fakeCollectionFactory();
+  const fakeAsyncOperation = fakeAsyncOperationFactory({
+    id: uuid(),
+    output: '{ "output": "test" }',
+  });
+
+  const [
+    existingExecution,
+    existingCollection,
+    existingAsyncOperation,
+  ] = await Promise.all([
+    executionsModel.create(fakeExecution),
+    collectionsModel.create(fakeCollection),
+    asyncOperationsModel.create(fakeAsyncOperation),
+  ]);
+
+  t.teardown(() => Promise.all([
+    executionsModel.delete({ arn: fakeExecution.arn }),
+    collectionsModel.delete(fakeCollection),
+    asyncOperationsModel.delete({ id: fakeAsyncOperation.id }),
+  ]));
+
+  // migrate existing async operation and collection
+  await migrateAsyncOperationRecord(existingAsyncOperation, t.context.knex);
+  await migrateCollectionRecord(existingCollection, t.context.knex);
```

Comment:
I could be wrong, but I'm not sure that we need to create the async operation or collection in Dynamo for the purposes of testing this Lambda. We only need them to exist in Postgres so they can be referenced by the execution via foreign keys, correct?

Separately, I would like to avoid using `migrateAsyncOperationRecord` and `migrateCollectionRecord` in the tests here. The reason why is that now the tests for data-migration2 are dependent on the code for data-migration1, so bugs in data-migration1 code could cause both tests to fail. Instead, I'd probably opt to just create any objects that you need to reference via FKs directly via the models and using these helpers:

https://github.com/nasa/cumulus/blob/feature-rds/packages/db/src/test-utils.ts#L50

Thoughts?

Reply:
I think that's totally right. I'm working on this refactor now + setting up a mock factory for async operations. Originally I was thinking we would need the dynamo records for Collections and Async Operations because we need to set a couple fk fields on the _dynamo_ execution:

```
  // Create new Dynamo execution to be migrated to postgres
  const newExecution = fakeExecutionFactoryV2({
    parentArn: existingExecution.arn,
    collectionId: `${fakeCollection.name}___${fakeCollection.version}`,
    asyncOperationId: fakeAsyncOperation.id,
  });
```
But all of those fields from collections and async operations are also present in the PG table so there's no reason to jump through these hoops and create/migrate Dynamo records.

Thanks for pointing this out. It'll be much cleaner.

Reply:
Updated the PR. We're no longer creating collections/async operations in dynamo in this test. There's also a new `fakeAsyncOperationRecordFactory` to create the PG record for async operations.